iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/ResourceDeclarationConverter.java

```diff
@@ -69,6 +75,26 @@ public ResourceDeclaration convertToResourceDeclaration(MappingTree tree) {
       .map(this::toProperties)
       .orElse(Collections.emptyList());
 
-    return new ResourceDeclarationImpl(name, version, type, otherProperties);
+    Optional<PropertyTree> optChildResources = PropertyUtils.get(tree, "resources");
+    if (optChildResources.isPresent()) {
+      List<ResourceDeclaration> childResources = optChildResources
+        .map(PropertyTree::value)
+        .filter(SequenceTree.class::isInstance)
+        .map(SequenceTree.class::cast)
+        .map(sequenceTree -> mappingTreeOnly(sequenceTree.elements()))
+        .map(m -> m.stream().map(mappingTree -> convertToResourceDeclaration(mappingTree, buildParentType(parentType, type))).collect(Collectors.toList()))
+        .orElse(Collections.emptyList());
+      return new GroupResourceDeclarationImpl(name, version, type, parentType, otherProperties, childResources);
+    } else {
+      return new ResourceDeclarationImpl(name, version, type, parentType, otherProperties);
+    }
+  }
+
+  private static String buildParentType(@Nullable String parentType, StringLiteral type) {
+    if (parentType == null) {
+      return type.value();
+    } else {
+      return parentType + "/" + type.value();
+    }
```

Comment:
I would not reference the parent type as a string. You could simply put the parent resource as an object reference. That we could also access TextRanges and other properties of the parent resource.